docs/documentation/cert-manager.md

```diff
@@ -0,0 +1,198 @@
+
+```
+dekorate.certificate.secret-name=tls-secret
+# The selfSigned issuer:
+dekorate.certificate.self-signed.enabled=true
+```
+
+Or via the `@Certificate` annotation:
+
+```java
+@Certificate(secretName = "tls-secret", selfSigned = @SelfSigned(enabled = true))
+public class Main {
+    // ...
+}
+```
+
+This configuration will generate up to two resources under `target/classes/dekorate/kubernetes.yml` file that should look like this:
+
+```
+---
+apiVersion: cert-manager.io/v1
+kind: Issuer
+metadata:
+  name: kubernetes-example
+spec:
+  selfSigned: {}
+---
+apiVersion: cert-manager.io/v1
+kind: Certificate
+metadata:
+  name: kubernetes-example
+spec:
+  encodeUsagesInRequest: false
+  isCA: false
+  issuerRef:
+    name: kubernetes-example
+  secretName: tls-secret
+```
+
+Apart from these two resources, the Cert-Manager Dekorate extension will also configure, part of the Deployment, a volume mounted from the secret that contain the Cert-Manager generated files to allow the application to access them and to configure the HTTPS/TLS endpoint:
+
+```
+---
+apiVersion: apps/v1
+kind: Deployment
+metadata:
+  name: kubernetes-example
+spec:
+  replicas: 1
+  template:
+    spec:
+      containers:
+        - name: kubernetes-example
+          volumeMounts:
+            - mountPath: /etc/certs
+              name: volume-certs
+              readOnly: true
+      volumes:
+        - name: volume-certs
+          secret:
+            optional: false
+            secretName: tls-secret
+```
+
+#### Usage
+
+To let the application consume the path mounted, we have then to update also the properties of the application (Quarkus, Spring Boot, ...) to use Cert-Manager the generated files that are mounted in the folder `/etc/certs`.
+
+To see a practical working example, please go to [the Spring Boot with Cert-Manager](https://github.com/dekorateio/dekorate/tree/main/examples/spring-boot-with-certmanager-example) example which uses a PKCS keystore.
+
+#### Securing Resources
+
+When securing your resources, it's important to validate that the requests are coming from known host names. For this purpose, we can use the `dnsNames` property which is part of the certificate configuration. For example, by adding the following `dekorate.certificate.dnsNames` property (it's a comma separated list of strings):
+
+```
+dekorate.certificate.dnsNames=foo.bar.com
+```
+
+The certificate will only allow requests coming from the host `foo.bar.com`.
+
+Note that the applications in Kubernetes can be publicly exposed using [Ingress](https://kubernetes.io/docs/concepts/services-networking/ingress/) resources, for example:
+
+```yaml
+apiVersion: networking.k8s.io/v1
+kind: Ingress
+metadata:
+  name: kubernetes-example
+spec:
+  rules:
+  - host: "foo.bar.com"
+    http:
+      paths:
+      - pathType: Prefix
+        path: "/"
+        backend:
+          service:
+            name: kubernetes-example
+            port:
+              number: 8080
```

Comment:
The TLS section and TLS secret should be added as we expose a HTTP(S) endpoint ;-)

Reply:
It seems we can't configure this section in Dekorate. I've added a comment in the documentation and reported an issue to implement this: https://github.com/dekorateio/dekorate/issues/967